src/hooks/useVideosAPI.js

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+import { youtubevideos } from '../mocks/youtubevideos';
+require('dotenv').config();
+
+const useVideoApi = () => {
+    const [videos, setVideos] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState('');
+
+    const fetchVideos = async (searchText) => {
+        setLoading(true);
+        try{
+            const YOUTUBE_API_KEY = 'AIzaSyBPRIGYlC55KfMs3_Xy_Y5An0-g1cH5y1w';
```

Comment:
Is recommended to put as env variable the youtube api key in order to have more flexibility when you change the api key value